src/Application.php

```diff
@@ -70,6 +90,33 @@ public function __construct(ContainerBuilder $containerBuilder = null, array $va
                 $this['phpdi.callable_resolver']
             );
         });
+
+        // Override the dispatcher with ours to use our event listeners
+        $this['dispatcher'] = $this->share(function () {
+            /**
+             * @var EventDispatcherInterface
+             */
+            $dispatcher = new $this['dispatcher_class']();
+
+            $urlMatcher = new LazyUrlMatcher(function () {
+                return $this['url_matcher'];
+            });
+            if (Kernel::VERSION_ID >= 20800) {
+                $dispatcher->addSubscriber(new RouterListener($urlMatcher, $this['request_stack'], $this['request_context'], $this['logger']));
+            } else {
+                $dispatcher->addSubscriber(new RouterListener($urlMatcher, $this['request_context'], $this['logger'], $this['request_stack']));
+            }
+            $dispatcher->addSubscriber(new LocaleListener($this, $urlMatcher, $this['request_stack']));
+            if (isset($this['exception_handler'])) {
+                $dispatcher->addSubscriber($this['exception_handler']);
+            }
+            $dispatcher->addSubscriber(new ResponseListener($this['charset']));
+            $dispatcher->addSubscriber(new MiddlewareListener($this, $this->callbackInvoker));
+            $dispatcher->addSubscriber(new ConverterListener($this['routes'], $this['callback_resolver'], $this->callbackInvoker));
+            $dispatcher->addSubscriber(new StringToResponseListener());
+
+            return $dispatcher;
+        });
```

Comment:
I feel the same. I tried to implement this by getting all listeners with `getListeners()` and iterating through them, to find the Silex middleware subscriber, remove that and add our own. But that was more code in the end, and less performant for sure. I also don't know if the changed order of subscribers would cause any problem. It is unfortunate, but this is the problem everytime you overwrite something in OOP, you always duplicate some implementation.